content/blog/custom-docker-images-for-emr-serverless.md

```diff
@@ -0,0 +1,199 @@
+
+In this article we take two scenarios , one where the application expects the data jars to be in specific location for its execution as in case of Delta lake and another where we install specific python modules using pip install both of which are bundled into a custom docker image that the EMR serverless uses. 
+
+Delta lake (<https://delta.io>) is an open-source storage framework that enables building a lake house architecture. One of the important features supported by delta lake is the checkpoint management for batch/streaming jobs that uses AWS S3 as a data lake. The most common approach to use Delta lake is via command line argument to run spark-submit command something like *"sparkSubmitParameters": "--packages io. delta:delta-core_2.12:1.2.1"*. This dynamically pulls the delta lake (via maven project) when the spark jobs run (ref. <https://docs.delta.io/latest/quick-start.html#pyspark-shell> ). The other approach is to use **pip install delta-spark==2.2.0** when python is used**.** However, both these approaches in production might not be possible as they require an active internet connection, necessary permissions on the production machine to download/install a library at run time in the EMR serverless application.
+
+Our second use case is the installation of all custom libraries (in the form of jars etc.,) used by the code. We provide step by step walkthrough of how we build custom images addressing both the above scenarios, use the ECR to register our docker image, and finally use the docker image in the EMR Serverless.
+
+*Step 1:*  Pre-requisites
+
+*Step 2:*  Identify the base EMR Serverless Docker Image, packages/jars that need to be installed/downloaded, and prepare the custom docker Image.
+
+*Step 3:*  Push the Docker Image to the ECR
+
+*Step 4:* Create the EMR Serverless applications via
+
+(a) Command using AWS CLI
+
+(b) through settings in the UI when creating or updating the EMR serverless application.
+
+## Step 1: Prerequisites to build a custom docker image:
+
+* The latest version of the docker client
+* The latest version of AWS CLI
+* Verify the access to dependent repository from where resources (modules, code etc) might be downloaded.   
+
+## **Step 2: Sample Docker file used to create custom EMRServerless image**
+
+<!--StartFragment-->
+
+*\# Refer AWS EMR documentation for the release version tag*\
```

Comment:
It is recommended to use "code block snippet" for code block. 
For example, you can use the code block snippet with language syntax: for example "bash" "shell" or "markdown".
You can use the CMS editor mode "Rich text" to add code block and specify appropriate language syntax.

Please refer to the guideline section [here](https://github.com/hpe-dev-incubator/hpe-dev-portal/blob/master/docs/ContributorGuide-v2.md#tips-and-trick-using-the-cms-editor), Tip _10_.